packages/backend/src/apps/postman/common/throw-errors.ts

```diff
@@ -1,59 +1,105 @@
   error: HttpError,
-  position: number,
-  appName: string,
-): never {
-  const postmanErrorData: PostmanApiErrorData = error.response.data
+): PostmanEmailSendStatus {
+  const postmanErrorData: PostmanApiErrorData = error.response?.data ?? {}
   const { code: errorCode, message: errorMessage } = postmanErrorData
   // catch common postman error codes and provide solution
   switch (errorCode) {
     case 'invalid_template':
-      if (errorMessage.includes('attachments')) {
-        throw new StepError(
-          'Unsupported attachment file type',
-          'Click on set up action and check that the attachment type is supported by postman. Please check the supported types at [this link](https://guide.postman.gov.sg/email-api-guide/programmatic-email-api/send-email-api/attachments#list-of-supported-attachment-file-types).',
-          position,
-          appName,
-          error,
-        )
-      } else if (errorMessage.includes('blacklisted')) {
-        throw new StepError(
-          'Blacklisted recipient email',
-          'Please request for the recipient email to be whitelisted at [this link](https://guide.postman.gov.sg/api-guide/programmatic-email-api/send-email-api/recipient-blacklist).',
-          position,
-          appName,
-          error,
-        )
+      if (errorMessage?.includes('attachments')) {
+        return 'INVALID-ATTACHMENT'
+      } else if (errorMessage?.includes('blacklisted')) {
+        return 'BLACKLISTED'
       } else {
         // return original error if not caught
-        throw error
+        return 'ERROR'
       }
     case 'rate_limit':
+      return 'RATE-LIMITED'
+    default:
+      // return original error if not caught
+      return 'ERROR'
+  }
+}
+
+export function throwPostmanStepError({
+  status,
+  position,
+  appName,
+  error,
+  isPartialSuccess,
+  blacklistedRecipients,
+}: {
+  status: PostmanEmailSendStatus
+  position: number
+  appName: string
+  error: HttpError
+  isPartialSuccess: boolean
+  blacklistedRecipients: string[]
+}) {
+  switch (status) {
+    case 'BLACKLISTED': {
+      const name = 'Blacklisted recipient email'
+      const solution = `The following email addresses have been blacklisted by Postman:
+         \n${blacklistedRecipients
+           .map((recipient) => `**${recipient}**`)
+           .join('\n\n')}
+         \nIf you believe that they are valid and active, please [contact the Postman team](https://go.gov.sg/postman-contact-us) to remove them from the blacklist and try again.
+        `
+      if (isPartialSuccess) {
+        throw new PartialStepError({
+          name,
+          solution,
+          position,
+          appName,
+          partialRetry: {
+            buttonMessage: 'Resend to blacklisted recipients',
+          },
+        })
+      }
+      throw new StepError(name, solution, position, appName, error)
+    }
+    case 'RATE-LIMITED':
+      // this will be auto-retried later on
       throw new StepError(
```

Comment:
This will cause auto-retries to retry all 

I think 2 ways easy to address:
1. Throw PartialStepError, like the blacklisted case:
```suggestion
      if (isPartialSuccess) {
        throw new PartialStepError({
          name,
          solution,
          position,
          appName,
          partialRetry: null // new feature: null means system is auto retrying? 
        })
      }
```
2. In the main code, we never retry ACCEPTED